bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -1559,9 +1551,67 @@ impl TryFrom<&TokenTagDto> for TokenTag {
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct TokenSchemeDto {
+pub enum TokenSchemeDto {
+    /// A simple token scheme.
+    Simple(SimpleTokenSchemeDto),
+}
+
+impl From<&TokenScheme> for TokenSchemeDto {
+    fn from(value: &TokenScheme) -> Self {
+        match value {
+            TokenScheme::Simple(v) => Self::Simple(v.into()),
+        }
+    }
+}
+
+impl TryFrom<&TokenSchemeDto> for TokenScheme {
+    type Error = Error;
+
+    fn try_from(value: &TokenSchemeDto) -> Result<Self, Self::Error> {
+        Ok(match value {
+            TokenSchemeDto::Simple(v) => Self::Simple(v.try_into()?),
+        })
+    }
+}
+
+/// Describes a foundry output that is controlled by an alias.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SimpleTokenSchemeDto {
     #[serde(rename = "type")]
     pub kind: u8,
+    // Amount of tokens minted by a foundry.
+    #[serde(rename = "mintedTokens")]
+    pub minted_tokens: U256Dto,
+    // Amount of tokens melted by a foundry.
+    #[serde(rename = "meltedTokens")]
+    pub melted_tokens: U256Dto,
+    // Maximum supply of tokens controlled by a foundry.
+    #[serde(rename = "maximumSupply")]
+    pub maximum_supply: U256Dto,
+}
+
+impl From<&SimpleTokenScheme> for SimpleTokenSchemeDto {
+    fn from(value: &SimpleTokenScheme) -> Self {
+        Self {
+            kind: SimpleTokenScheme::KIND,
+            minted_tokens: U256Dto(value.minted_tokens().to_string()),
+            melted_tokens: U256Dto(value.melted_tokens().to_string()),
+            maximum_supply: U256Dto(value.maximum_supply().to_string()),
```

Comment:
I know it's not part of this PR, but do the `U256Dto` fields have the right encoding? For `minted_tokens`, `melted_tokens`, and `maximum_supply` we simply use `to_string` on `U256`. And for `NativeTokens` we use `serde`.

Shouldn't all of these use `prefix_hex`? I might have missed that on my PR.

Reply:
May I ask you to change this in a a subsequent PR ?

Reply:
Sure, no worries.